javascript/ql/src/semmle/javascript/Functions.qll

```diff
@@ -204,6 +204,9 @@ class Function extends @function, Parameterized, TypeParameterized, StmtContaine
   /** Holds if this function is an asynchronous function. */
   predicate isAsync() { isAsync(this) }
 
+  /** Holds if this function is not asynchronous and also not a generator. */
+  predicate isOrdinary() { not isAsync() and not isGenerator() }
```

Comment:
I'd say there are too many kinds of `Function` instances to have a predicate named `isOrdinary`. Implicit class constructors, abstract methods, bodiless functions in ambient context, function signatures etc, all seem non-ordinary in their own way.

I'd suggest `isAsyncOrGenerator` and then replacing the calls with `not f.isAsyncOrGenerator()`